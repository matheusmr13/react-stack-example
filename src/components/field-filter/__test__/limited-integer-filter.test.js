import React from 'react';
import LimitedIntegerFilter from 'components/field-filter/limited-integer-filter';
import { shallow } from 'enzyme';

describe('LimitedIntegerFilter component', () => {
	it('should render properly', () => {
		const spec = {
			id: 'myId',
			name: 'Limited integer field',
			validation: {
				min: 0,
				max: 10
			}
		};
		const limitedIntegerFilter = shallow(<LimitedIntegerFilter spec={spec} />);
		expect(limitedIntegerFilter).toMatchSnapshot();
	});
});
