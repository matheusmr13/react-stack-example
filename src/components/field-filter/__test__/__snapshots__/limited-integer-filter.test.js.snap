// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LimitedIntegerFilter component should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LimitedIntegerFilter
    onChange={[Function]}
    spec={
        Object {
            "id": "myId",
            "name": "Limited integer field",
            "validation": Object {
              "max": 10,
              "min": 0,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "axis": "x",
      "defaultValue": 5,
      "disableFocusRipple": false,
      "disabled": false,
      "max": 10,
      "min": 0,
      "onChange": [Function],
      "required": true,
      "step": 1,
      "style": Object {},
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "axis": "x",
        "defaultValue": 5,
        "disableFocusRipple": false,
        "disabled": false,
        "max": 10,
        "min": 0,
        "onChange": [Function],
        "required": true,
        "step": 1,
        "style": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
